### DISCUSSION TITLE
请问如何根据域名实现重定向啊

### DISCUSSION THREAD

**Discussion by ghost** [CaiJimmy/hugo-theme-stack#361] [Q&A] [answered]
我最近把项目部署到了firebase，他给了我两个贼挫的域名，又不能取消，所以我想把这两个挫域名重定向到我自己的域名上，有什么办法实现吗？谢谢！๑•́₃•̀๑

比如根据访问的域名重定向这种

**Comment by zhixuan2333** [comment, marked as answer]
如果你是想添加自己的域名 -> https://firebase.google.com/docs/hosting/custom-domain?hl=zh-cn

**Reply by ghost** [reply]
域名已经添加了，现在我的web有三个域名同时存在，两个是他的默认域名，第三个就是我自己的，总感觉这样不太好。

所以我想如果访问的是默认域名（xxx.web.app这种），就重定向到我自己的域名（blog.xxx.com)。

**Reply by zhixuan2333** [reply]
别人应该是不会知道URL吧，vercel,netlify也是这样的。

**Reply by ghost** [reply]
我在网上只搜到一个人说过这个问题，但是不详细。

就是请求的时候判断访问的域名是啥，如果不是我自己的域名，就重定向到我自己的域名。

但是我不会添加，不知道该加到哪里，应该是用js实现的。

**Reply by ghost** [reply]
> 别人应该是不会知道URL吧，vercel,netlify也是这样的。

总觉得别扭，而且网上说多个域名内容一样，也会影响收录

**Reply by zhixuan2333** [reply]
js应该是不不行的吧 BOT 应该不会执行JS的

**Reply by ghost** [reply]
好像是的...那有其他方法了吗

**Reply by zhixuan2333** [reply]
在head 里面加  `<meta name="robots" content="noindex,follow" />` 之类的。

**Reply by ghost** [reply]
这三个域名指向的同一个工程，不通过脚本的话，一改全改了😂😂

**Reply by zhixuan2333** [reply]
😅

**Reply by ghost** [reply]
我看到有人说，可以在http上加 
```
<link rel="canonical" href="http://myapp.domain.com/" />
```
，那么要这么做的话，需要添加到哪里呢

**Reply by zhixuan2333** [reply]
![image](https://user-images.githubusercontent.com/59254886/134444442-b8e4927c-ee63-4fd0-9547-03753fb73a0f.png)
主题自带

**Reply by ghost** [reply]
我搜了下我的，没有，是不是我缺了啥配置🤣🤣

**Reply by zhixuan2333** [reply]
是不是自定义了 `layouts/partials/head/head.html`?

**Reply by zhixuan2333** [reply]
在 SEO 里面 `http://example.com/hello/` != `http://example.com/hello/index.html`
如果加 `<link rel="canonical" href="http://example.com/hello/" />`

不管是
```
http://example.com/hello/
http://example.com/hello/?q=1
http://example.com/hello/index.html
...
```
在SEO里都会指向 `http://example.com/hello/`

**Reply by ghost** [reply]
我从已经部署好的页面，看到了这个标签，但是依旧没有自动跳转，不清楚他们说的这个标签具体有什么作用

**Reply by ghost** [reply]
明白了，谢谢！
